cmd/notation/verify.go

```diff
@@ -45,120 +41,49 @@ func verifyCommand(opts *verifyOpts) *cobra.Command {
 			return runVerify(cmd, opts)
 		},
 	}
-	setFlagSignature(command.Flags(), &opts.signatures)
-	command.Flags().StringSliceVarP(&opts.certs, "cert", "c", []string{}, "certificate names for verification")
-	command.Flags().StringSliceVar(&opts.certFiles, cmd.PflagCertFile.Name, []string{}, "certificate files for verification")
-	command.Flags().BoolVar(&opts.pull, "pull", true, "pull remote signatures before verification")
 	opts.ApplyFlags(command.Flags())
+	command.Flags().StringSliceVar(&opts.config, "config", nil, "verification plugin config (accepts multiple inputs)")
 	return command
 }
 
 func runVerify(command *cobra.Command, opts *verifyOpts) error {
-	// initialize
+	// initialize.
 	verifier, err := getVerifier(opts)
 	if err != nil {
 		return err
 	}
-	manifestDesc, err := getManifestDescriptorFromContext(command.Context(), &opts.RemoteFlagOpts, opts.reference)
-	if err != nil {
-		return err
-	}
 
-	sigPaths := opts.signatures
-	if len(sigPaths) == 0 {
-		if !opts.Local && opts.pull {
-			if err := pullSignatures(command, opts.reference, &opts.SecureFlagOpts, digest.Digest(manifestDesc.Digest)); err != nil {
-				return err
-			}
-		}
-		manifestDigest := digest.Digest(manifestDesc.Digest)
-		sigDigests, err := cache.SignatureDigests(manifestDigest)
-		if err != nil {
-			return err
-		}
-		for _, sigDigest := range sigDigests {
-			sigPaths = append(sigPaths, dir.Path.CachedSignature(manifestDigest, sigDigest))
+	// set up verification plugin config.
+	configs := make(map[string]string)
+	for _, c := range opts.config {
+		parts := strings.Split(c, ":")
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid config option: %s", c)
 		}
+		configs[parts[0]] = parts[1]
 	}
+	ctx := command.Context()
+	verification.WithPluginConfig(ctx, configs)
```

Comment:
Where do we use the return value?

Reply:
configs would be retrieved via method: https://github.com/notaryproject/notation-go/blob/dev-rc.1/verification/types.go#L225 in https://github.com/notaryproject/notation-go/blob/dev-rc.1/verification/verifier_helpers.go#L299

Reply:
> configs would be retrieved via method: https://github.com/notaryproject/notation-go/blob/dev-rc.1/verification/types.go#L225 in https://github.com/notaryproject/notation-go/blob/dev-rc.1/verification/verifier_helpers.go#L299

But `context.WithValue` will create a new context instead of putting value in the current context.
https://pkg.go.dev/context#WithValue
It is more like a linked list.

Reply:
Fixed it. thanks!